packages/web_ui/src/components/ModPackViewPage.tsx

```diff
@@ -424,7 +424,15 @@ function InputColor(props: {
 	value: lib.ModSettingColor,
 	onChange: (value: lib.ModSettingColor) => void,
 }) {
-	const value = props.value;
+	// In certain edge cases it's possible a value for this mod setting does not exist
+	// or is of the wrong type. Fallback to black if this is the case.
+	function isColor(input: unknown) {
+		if (typeof input !== "object" || input === null) {
+			return false;
+		}
+		return "r" in input && "g" in input && "b" in input && "a" in input;
```

Comment:
Not quite sure where this comes from, but isn't alpha always optional, defaulting to 1?

Reply:
In the data stored to mod-settings.dat by the game alpha is always present. The values comes from `ModPack.fillDefaultSettings` which sources them from settings prototypes and normalize them with `normalizeColor` which makes sure all components are always present.